bootstrap/secret/secure.go

```diff
@@ -349,3 +349,32 @@ func prepareSecret(secret ServiceSecret) (string, map[string]string) {
 
 	return path, secretsKV
 }
+
+// HasSecret returns true if the service's SecretStore contains a secret at the specified path.
+func (p *SecureProvider) HasSecret(path string) (bool, error) {
+	if cachedSecrets := p.getSecretsCache(path); cachedSecrets != nil {
+		return true, nil
+	}
+
+	if p.secretClient == nil {
+		return false, errors.New("can't get secrets. Secure secret provider is not properly initialized")
+	}
+
+	secureSecrets, err := p.secretClient.GetSecrets(path)
+
+	retry, err := p.reloadTokenOnAuthError(err)
+	if retry {
+		// Retry with potential new token
+		secureSecrets, err = p.secretClient.GetSecrets(path)
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	if secureSecrets == nil {
```

Comment:
Yes, I agree.  The golang idiom is that if there is an error, nil objects are returned, and vice-versa.  Checking for a nil object is tantamount to checking for err != nil.   I think the resulting code will be extremely fragile if we choose to inspect the error message for a particular string.  What would be better would be to return a typed error with the error code in it and check for 404 locally, or call another method that explicitly checks for a 404, as Lenny says.

Reply:
go-mod-secrets' code could be enhanced to return a new typed for SecretNameNotFound when a 404 is retuned from Vault.

FYI, in 3.0 I would like to change `path` to `name` since that is how it is used.

Reply:
> FYI, in 3.0 I would like to change `path` to `name` since that is how it is used.

Better but that in the 3.0 project board because surely I'll forget.